### ISSUE TITLE
Helper for Building pdata.Metrics

### ISSUE THREAD

**Issue by keitwb** [open-telemetry/opentelemetry-collector#1554]
`pdata.Metrics`, the new metric data strcture that will be standard soon, is fairly hard to use directly.  There are many levels in the hierarchy and it is impossible to instantiate anything using struct literals and it is generally quite verbose and requires a very imperative coding style to do anything significant with it.  Most (if not all) of the existing metrics functionality in the collector uses the OpenCensus protobuf structures, which is easier to work with but is going to be non-standard soon.

My team is going to be focused over the next few months on adding a lot of metrics functionality to the collector, especially around adding receivers that can connect to external services and generate metrics from monitoring data emitted by them (like the [Redis receiver in contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/master/receiver/redisreceiver)).  Given the verbosity of working with `pdata.Metrics` I foresee a lot of "util" type functions being written within each receiver to instantiate metrics, so it seems better to just write a high-quality version of those up front so that people writing the receivers can focus on the logic of the integration.

Here are some desirable features:

 - Ability to tersely add a new datapoint with a certain metric name, set of labels, and value (e.g. `AddGauge("my-gauge", map[string]string{"label-a": "value-a"}, 5)`).  Receivers should not have to worry about aggregating datapoints into `pdata.*DataPointSlice`s with common descriptors.
 - For a given receiver, the `pdata.InstrumentationLibrary` will generally be the same for all datapoints, so receivers should be able to just specify that declaratively up front.
 - Handles timestamps conveniently (usually all datapoints generated in a given "cycle" will have the same timestamp) using standard Go `time.Time` struct instead of `TimestampUnixNano`.
 - Handles Resource attributes conveniently (generally this will be the same for all datapoints within a certain receiver configuration) (i.e. using string->string maps)
 - Handles labels in a Go-friendly manner (i.e. using string->string maps) and in a way that allows the specification of labels up front for a large set of metrics
 - Handles all six types of metrics (summary and histogram are the hardest since they are themselves complex structures)

Basically the overarching goal is just to make receivers have to have the minimum readable/understandable code necessary to do what they need conveniently without adding too much abstraction on top of `pdata.Metrics`.

I have written a `MetricsBuilder` struct in #1540 as an attempt to make `pdata.Metrics` much easier to work with.

Any other requirements or feedback on the PR are welcome.
